app/lib/cms/sts/assume_role_web_identity.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module CMS
+  module STS
+    class AssumeRoleWebIdentity
+      include Singleton
+      # :reek:Attribute
+      attr_accessor :region, :role_arn, :role_session_name, :web_identity_token_file
+
+      class TokenNotFoundError < StandardError
+      end
+
+      def identity_credentials
+        @identity_credentials ||= begin
+          validate_configuration
+
+          Aws::AssumeRoleWebIdentityCredentials.new(
+            region: region,
+            role_arn: role_arn,
+            role_session_name: role_session_name,
+            web_identity_token_file: web_identity_token_file
+          )
+        end
+      end
+
+      private
+
+      attr_reader :params
+
+      def validate_configuration
```

Comment:
Not a big deal again but "validation" is a bit misleading. This only checks the token presence, not its validity.

Reply:
I understood it as validating that configuration options are set. But given it does only that - check token presence, what you say actually makes sense!

Reply:
Changed in 18c48a2.